unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/FeatureController.cs

```diff
@@ -23,7 +24,14 @@ public void Start()
         KernelConfig.i.OnChange += OnKernelConfigChanged;
     }
 
-    // Update is called once per frame
+    public void OnGUI()
```

Comment:
If we are using `OnGUI` only for the selection box, maybe we can extract the selection object to a specific `MonoBehaviour` to avoid exposing `OnGUI` to all the features? I think this can help to mitigate further usages, as this event is pretty controversial.

Reply:
We just have removed all the monobehaviours from builder in world to try to remove full dependency on them.

Making this change sounds like a hack in the current implementation to me. I totally understand your desire to remove That part but it is more obscure to encapsulate that call in a monobehaviour.

I will rather add it as a technical debt and change it in the future rather than that. Anyway, your call!